core/server/models/post.js

```diff
@@ -185,6 +189,52 @@ Post = ghostBookshelf.Model.extend({
         model.emitChange('deleted', Object.assign({usePreviousAttribute: true}, options));
     },
 
+    onDestroying: function onDestroyed(model) {
+        this.handleAttachedModels(model);
+    },
+
+    handleAttachedModels: function handleAttachedModels(model) {
+        model.related('tags').on('detaching', function onDetached(collection, tag) {
+            model.related('tags').on('detached', function onDetached(detachedCollection, response, options) {
+                tag.emitChange('detached', options);
+            });
+        });
+
+        model.related('tags').on('attaching', function onDetached(collection, tags) {
+            model.related('tags').on('attached', function onDetached(detachedCollection, response, options) {
+                tags.forEach(tag => tag.emitChange('attached', options));
+            });
+        });
+
+        model.related('authors').on('detaching', function onDetached(collection, author) {
+            model.related('authors').on('detached', function onDetached(detachedCollection, response, options) {
+                author.emitChange('detached', options);
+            });
+        });
+
+        model.related('authors').on('attaching', function onDetached(collection, authors) {
+            model.related('authors').on('attached', function onDetached(detachedCollection, response, options) {
+                authors.forEach(author => author.emitChange('attached', options));
+            });
+        });
+    },
+
+    /**
+     * @NOTE:
+     * when status is changed from or to 'published' all related authors and tags
+     * have to trigger recalculation in URL service because status is applied in filters for
+     * these models
+     */
+    handleStatusForAttachedModels: function handleStatusForAttachedModels(model, options) {
+        model.related('tags').forEach((tag) => {
+            tag.emitChange('attached', options);
```

Comment:
Don't have a strong opinion around naming this event anything different. The idea is to just trigger recalculation on URL services side and there is no change to the resource itself as such :man_shrugging:

Reply:
Re-using this event makes sense i think 👍